### DISCUSSION TITLE
Make organizing variables in inspector more intuitive and easy to use

### DISCUSSION THREAD

**Discussion by seppoday** [godotengine/godot-proposals#4149] [Editor]
Hello,

I was wondering if we can improve user experience when it comes to adding categories and groups for script variables in the inspector. Right now to get something like this:
![image](https://user-images.githubusercontent.com/62170071/156460344-09875446-2b88-4b5c-ae00-ad07a995c9a8.png)

You need to write code like this:
(which in my opinion is a lot of work, very hard to understand for beginners (its hard to follow docs) and it is just bad... you very often need to just copy same thing over and over again) and also alot of the times people also use _set and _get functions with it. Which makes it even more hard to read and write.
```
@tool
extends Node2D

@export var attack = 10
@export var my_name = "Name"
@export var can_attack = true

var jump_force = 200
var move_speed = 3

func _get_property_list() -> Array:
	var props = []
	props.append(
		{
			name = 'Character',
			type = TYPE_NIL,
			usage = PROPERTY_USAGE_CATEGORY | PROPERTY_USAGE_SCRIPT_VARIABLE
		}
	)
	props.append(
		{
			'name': 'Move',
			'type': TYPE_NIL,
			'hint_string': 'move_',
			'usage': PROPERTY_USAGE_GROUP | PROPERTY_USAGE_SCRIPT_VARIABLE
		}
	)
	props.append(
		{
			'name': 'move_speed',
			'type': TYPE_INT
		}
	)
	props.append(
		{
			'name': 'jump_force',
			'type': TYPE_INT
		}
	)
	return props

```

I was wondering if we can cut it down to something like this:

## IF YOU JUST WANT TO EXPORT VARIABLE
@export var my_name = "Name"

## IF YOU WANT TO MAKE CATEGORY
@export_category = "Characters"

## IF YOU WANT TO MAKE GROUP
@export_category = "Characters" (plus somehow add hints if you want??? and type)

## IF YOU WANT TO ADD VARIABLE TO CATEGORY OR GROUP
@export (Characters) var attack = 10

So above code could look something like that?

```
@tool
extends Node2D

@export var attack = 10
@export var my_name = "Name"
@export var can_attack = true

@export_category = "Characters"
@export (Characters) var jump_force = 200

@export_group ("move_") = "Move"
@export var move_speed = 3
```

So basically 1/4 of code for similar possibilities or something like that? I mean lets discuss it! :)

**Comment by Calinou** [comment]
There's already a proposal tracking this: https://github.com/godotengine/godot-proposals/issues/1255

**Reply by seppoday** [reply]
Oh damn... I even proposed similar solution. Sorry, I was looking for Issues/PR but didn't found it. Kinda shame that it is in discussion for 1.5 yrs, but sooner or later someone will implement it :)
